scripts/globals/follow.lua

```diff
@@ -56,30 +57,59 @@ function addFollower(leader, follower, options)
     local leaderId = leader:getID()
     local followerId = follower:getID()
 
-    if followerToLeaderMap[followerId] then
-        removeFollower(followerToLeaderMap[followerId], follower)
-    end
+    xi.follow.stopFollowing(follower)
 
     leaderToFollowersMap[leaderId] = leaderToFollowersMap[leaderId] or {}
 
     table.insert(leaderToFollowersMap[leaderId], follower)
     followerToLeaderMap[followerId] = leader
 
     follower:addListener("ROAM_ACTION", "FOLLOW_ROAM_ACTION", onMobRoamAction)
+    follower:addListener("DESPAWN", "FOLLOW_DESPAWN", onMobDespawn)
+    follower:setLocalVar("[follow]roamFlags", follower:getRoamFlags())
+    follower:setLocalVar("[follow]DONT_ROAM_HOME", follower:getMobMod(xi.mobMod.DONT_ROAM_HOME))
+    follower:setLocalVar("[follow]ROAM_COOL", follower:getMobMod(xi.mobMod.ROAM_COOL))
     follower:setRoamFlags(xi.roamFlag.SCRIPTED)
     follower:setMobMod(xi.mobMod.DONT_ROAM_HOME, 1)
-    follower:setMobMod(xi.mobMod.ROAM_COOL, options.roamCooldown)
+    if options.roamCooldown then
+        follower:setMobMod(xi.mobMod.ROAM_COOL, options.roamCooldown)
+    end
+
     followerOptions[followerId] = options
 end
 
-function removeFollower(leader, follower)
-    local leaderId = leader:getID()
+function cleanupFollower(follower)
+    followerOptions[follower:getID()] = nil
+    follower:removeListener("FOLLOW_ROAM_ACTION")
+    follower:removeListener("FOLLOW_DESPAWN")
+    follower:setRoamFlags(follower:getLocalVar("[follow]roamFlags"))
+    follower:setMobMod(xi.mobMod.DONT_ROAM_HOME, follower:getLocalVar("[follow]DONT_ROAM_HOME"))
+    follower:setMobMod(xi.mobMod.ROAM_COOL, follower:getLocalVar("[follow]ROAM_COOL"))
+end
+
+function xi.follow.clearFollowers(leader)
```

Comment:
Related to the comment above, can global functions that you edit follow this structure?

xi.follow.functionName = function(params)